pkg/controllers/management/rkeworkerupgrader/upgrade.go

```diff
@@ -62,6 +65,14 @@ func (uh *upgradeHandler) Sync(key string, node *v3.Node) (runtime.Object, error
 		if cluster.DeletionTimestamp != nil || cluster.Status.AppliedSpec.RancherKubernetesEngineConfig == nil {
 			return nil, nil
 		}
+
+		if cluster.Annotations[clusterprovisioner.RkeRestoreAnnotation] == "true" {
+			if err := uh.restore(cluster); err != nil {
```

Comment:
```suggestion
			return node, uh.restore(cluster)
```

Reply:
This is fundamental change. Should the handler still return the object on err?

Reply:
this logic is under our own custom key enqueue, and has no object here to return  https://github.com/rancher/rancher/blob/78f5953f7a2e2cafcfe3bc290ddd9fa14e36a5e4/pkg/controllers/management/rkeworkerupgrader/upgrade.go#L62

Reply:
I'm not sure I follow... 
`return node, err` still works here

Reply:
not a logic change, the object is nil so changing code to `return node, uh.restore(cluster)`